runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsPlugin.java

```diff
@@ -238,19 +243,46 @@ private void checkPlayerSpawned(PlayerSpawned event)
 			return;
 
 		PlayerIndicatorType playerIndicatorType = playerIndicatorsService.getPlayerIndicatorType(player);
-		if (playerIndicatorType == null || !playerIndicatorType.equals(PlayerIndicatorType.NON_CLAN_MEMBER))
+		if (playerIndicatorType == null || playerIndicatorType.equals(PlayerIndicatorType.FRIENDS_CHAT_MEMBERS) && !config.notifyFriendsChatMembersSpawning()
+				|| playerIndicatorType.equals(PlayerIndicatorType.FRIEND) && !config.notifyFriendsSpawning())
 			return;
 
 		// Only send notifications in PvP zones
 		if (client.getVar(Varbits.PVP_SPEC_ORB) != 1 && client.getVar(Varbits.IN_WILDERNESS) != 1 &&
-				client.getWorldType().stream().noneMatch(x -> x == WorldType.DEADMAN))
+				client.getWorldType().stream().noneMatch(x -> x == WorldType.DEADMAN) && !config.notifyInPvpSafezone())
 			return;
 
+		// Only send notifications if player is attackable
+		if (config.notifyOnlyAttackablePlayers())
+		{
+			int ownCombatLevel = Objects.requireNonNull(client.getLocalPlayer()).getCombatLevel();
+			int wildernessLevel;
+			int lowestAttackable = ownCombatLevel;
+			int highestAttackable = ownCombatLevel;
+
+			if (WorldType.isPvpWorld(client.getWorldType()))
+			{
+				lowestAttackable -= 15;
+				highestAttackable += 15;
+			}
+
+			if (client.getVar(Varbits.IN_WILDERNESS) == 1)
+			{
+				wildernessLevel = Integer.parseInt(Objects.requireNonNull(client.getWidget(WidgetInfo.PVP_WILDERNESS_LEVEL)).getText().split(" ")[1].split("<")[0]);
+				lowestAttackable -= wildernessLevel;
+				highestAttackable += wildernessLevel;
+			}
+
+			if (player.getCombatLevel() < lowestAttackable || player.getCombatLevel() > highestAttackable)
+			{
+				return;
+			}
+
```

Comment:
Move this to it's own function as `checkPlayerSpawned()` is already fairly complex. Also `Objects.requireNonNull(object)` can still throw a null-pointer exception and acts as a a [fast-fail](https://en.wikipedia.org/wiki/Fail-fast) alternative to a null pointer later on. So I've added null checks in place of those.

```
	private boolean isPlayerAttackable(Player player)
	{
		if (client.getLocalPlayer() == null)
			return false;

		int ownCombatLevel = client.getLocalPlayer().getCombatLevel();

		if (WorldType.isPvpWorld(client.getWorldType()))
			return player.getCombatLevel() >= (ownCombatLevel - 15) && player.getCombatLevel() <= (ownCombatLevel + 15);

		if (client.getVar(Varbits.IN_WILDERNESS) == 1)
		{
			Widget levelRangeWidget = client.getWidget(WidgetInfo.PVP_WILDERNESS_LEVEL);
			if (levelRangeWidget == null)
				return false;

			int wildLevel = Integer.parseInt(levelRangeWidget.getText().split(" ")[1].split("<")[0]);
			return player.getCombatLevel() >= (ownCombatLevel - wildLevel) && player.getCombatLevel() <= (ownCombatLevel - wildLevel);
		}

		return false;
	}
```